matchbox_demo/src/box_game.rs

```diff
@@ -168,13 +180,14 @@ pub fn move_cube_system(
         }
         v.y *= FRICTION;
 
-        // constrain velocity (this way allows for fast diagonal movement, but its just an example)
-        v.x = v.x.min(MAX_SPEED);
-        v.x = v.x.max(-MAX_SPEED);
-        v.y = v.y.min(MAX_SPEED);
-        v.y = v.y.max(-MAX_SPEED);
-        v.z = v.z.min(MAX_SPEED);
-        v.z = v.z.max(-MAX_SPEED);
+        // constrain velocity
+        let mag = (v.x * v.x + v.y * v.y + v.z * v.z).sqrt();
+        if mag > MAX_SPEED {
+            let factor = MAX_SPEED / mag;
+            v.x *= factor;
+            v.y *= factor;
+            v.z *= factor;
+        }
```

Comment:
```suggestion
        v = v.clamp_length_max(MAX_SPEED);
```
(not tested)

Reply:
Since v is of type `Velocity`, which is a self-defined struct, the method `clamp_length_max` does not exist.

Reply:
Ah, I see, thought it was a `glam::Vec3`. I guess it should be, but don't need to do it in this pr.